web/src/filter.ts

```diff
@@ -1122,6 +1128,13 @@ export class Filter {
                 },
             );
         }
+        if (term_types.length === 3 && _.isEqual(term_types, ["channel", "topic", "with"])) {
+            const channel_name = this.operands("channel")[0];
+            assert(channel_name !== undefined);
+            this._sub = stream_data.get_sub_by_name(channel_name);
+            assert(this._sub !== undefined);
+            return this._sub.name;
+        }
```

Comment:
This should be placed directly next to the basic channel/topic code in this function; it's very confusing to have it sandwiched here. Why isn't it just an extra case next to this?

 `            (term_types.length === 3 && _.isEqual(term_types, ["channel", "topic", "near"])) ||`

Reply:
updated